resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -496,7 +497,7 @@ private[yarn] class YarnAllocator(
       executorIdCounter += 1
       val executorHostname = container.getNodeId.getHost
       val containerId = container.getId
-      val executorId = executorIdCounter.toString
+      val executorId = (initialExecutorIdCounter + executorIdCounter).toString
```

Comment:
it seems a bit strange to me to "add" the Ids?
@vanzin @jerryshao

Reply:
The initial problem was that initialExecutorIdCounter is coming from the driver which is already stopped. Making this lazy solved this. The other integer is necessary because make it `lazy var` is not possible.

Reply:
I get the point of fix. But also it seems a little strange to me. 

Besides, do we really need to fix your issue? As I know the case here is not a normal one.

Reply:
Yeah, this is an issue only when the application is quite fast. Do you have concerns in general solving this or related the fix in the first commit? Asking it because pausing the user class thread would be definitely better as I've written below.